service/src/integrationTest/java/uk/gov/hmcts/reform/fpl/controllers/GeneratedOrderControllerTest.java

```diff
@@ -125,20 +144,63 @@ void shouldReturnErrorsWhenFamilymanNumberIsNotProvided() {
         assertThat(callbackResponse.getErrors()).containsExactly("Enter Familyman case number");
     }
 
-    @Test
-    void shouldTriggerOrderEventWhenSubmitted() throws Exception {
-        String expectedCaseReference = "19898989";
-        postSubmittedEvent(buildCallbackRequest());
+    @Nested
+    class Submitted {
+
+        @AfterEach
+        void resetInvocations() {
+            reset(notificationClient);
+        }
+
+        @Test
+        void submittedShouldNotifyAdminAndLAWhenNoRepresentativesNeedServing() throws Exception {
+            postSubmittedEvent(buildCallbackRequest());
+
+            verify(notificationClient).sendEmail(
+                eq(ORDER_NOTIFICATION_TEMPLATE_FOR_LA),
+                eq(LOCAL_AUTHORITY_EMAIL_ADDRESS),
+                eq(expectedOrderLocalAuthorityParameters()),
+                eq(CASE_ID));
+
+            verify(notificationClient).sendEmail(
+                eq(ORDER_NOTIFICATION_TEMPLATE_FOR_ADMIN),
+                eq("admin@family-court.com"),
+                eq(getExpectedParametersForAdminWhenNoRepresentativesServedByPost()), eq(CASE_ID));
+
+            verifyZeroInteractions(notificationClient);
+        }
+
+        @Test
+        void submittedShouldNotifyAdminAndLAWhenRepresentativesNeedServingByPost() throws Exception {
+            given(documentDownloadService.downloadDocument(anyString())).willReturn(PDF);
 
-        verify(notificationClient).sendEmail(
-            eq(ORDER_NOTIFICATION_TEMPLATE), eq(LOCAL_AUTHORITY_EMAIL_ADDRESS),
-            eq(expectedOrderLocalAuthorityParameters()), eq(expectedCaseReference));
+            postSubmittedEvent(buildCallbackRequestWithRepresentatives());
+
+            verify(notificationClient).sendEmail(
+                eq(ORDER_NOTIFICATION_TEMPLATE_FOR_LA),
+                eq(LOCAL_AUTHORITY_EMAIL_ADDRESS),
+                eq(expectedOrderLocalAuthorityParameters()),
+                eq(CASE_ID));
+
+            verify(notificationClient).sendEmail(
+                eq(ORDER_NOTIFICATION_TEMPLATE_FOR_ADMIN),
+                eq("admin@family-court.com"),
+                dataCaptor.capture(),
+                eq(CASE_ID));
+
+            MapDifference<String, Object> difference = verifyNotificationSentToAdminWhenOrderIssued(dataCaptor,
+                IssuedOrderType.GENERATED_ORDER);
+
+            assertThat(difference.areEqual()).isTrue();
```

Comment:
why do we need to do this? Will assertThat(dataCaptor).isEqualTo(GENERATED_ORDER) not work? I cannot see the verifyNotificationSentToAdminWhenOrderIssued method so I am not sure of implementation. Maybe just explain to me what this is doing

Reply:
verifyNotificationSentToAdminWhenOrderIssued returns MapDifference between the actual data and the expected parameters (all found in NotifyAdminOrderIssuedTestHelper), so here we check that there's no difference (i.e we got what we expected) - nothing to do with GENERATED_ORDER, this is an enum I pass in so that I know if it's a CMO or not (cmo needs callout)